### PR TITLE
Bug 1906134: Don't create OperatorConditions for copied CSVs

### PR DESCRIPTION
**Pull request by awgreene**
Problem: OLM is currently creating an OperatorCondition for copied CSVs.
As a result, OLM is reporting that the deployments defined in the copied
CSVs cannot be found.

Solution: OLM should not be creating OperatorCondtions for copied CSVs.
Update the OperatorConditionGenerator Reconciler to include predicate
functions that exclude copied CSVs.

### CODE REVIEW

File: pkg/controller/operators/operatorconditiongenerator_controller.go [hunk 1]
```diff
@@ -34,9 +37,35 @@ func (r *OperatorConditionGeneratorReconciler) SetupWithManager(mgr ctrl.Manager
...
+			if _, ok := e.Meta.GetLabels()[operatorsv1alpha1.CopiedLabelKey]; ok {
+				return false
+			}
+			return true
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			if _, ok := e.MetaOld.GetLabels()[operatorsv1alpha1.CopiedLabelKey]; ok {
+				return false
+			}
+			return e.ObjectOld != e.ObjectNew
```
Comment (benluddy) [inline pkg/controller/operators/operatorconditiongenerator_controller.go:57]: Does `!=` have the right semantics for this?

Comment (awgreene) [reply pkg/controller/operators/operatorconditiongenerator_controller.go:57]: I had copied this bit from the [example code](controller ) in kubebuilder, I'm fine returning true though.

Comment (benluddy) [reply pkg/controller/operators/operatorconditiongenerator_controller.go:57]: Yeah, I suppose even if for whatever reason you have two value-equal objects for which `!=` evaluates to false, nothing bad happens. Returning `true` should be practically the same, and might save people like me from pausing to think about it, but I'm good with it either way. Thanks!

Comment (awgreene) [reply pkg/controller/operators/operatorconditiongenerator_controller.go]: In the interest of improved clarity I am updating this return statement to always return true.

**Comment by kevinrizza** [comment]
Tests?

**Comment by awgreene** [comment]
>Tests?

I can create a test that ensures an OpeneratorCondition wasn't created for a Copied CSV over x seconds but there is not a way to distinguish that the controller simply hasn't reconciled the Copied CSV yet.

**Comment by awgreene** [comment]
@kevinrizza I added the test described above which at least sounds an alarm if a Copied CSV is reconciled.

**Pull request merged**